numba/tests/test_extending.py

```diff
@@ -866,6 +866,41 @@ def impl(a, **kws):
         self.assertIn("use of VAR_KEYWORD (e.g. **kwargs) is unsupported", msg)
         self.assertIn("offending argument name is '**kws'", msg)
 
+    def test_overload_method_kwargs(self):
+        # Issue #3489
+        @overload_method(types.Array, 'foo')
+        def fooimpl(arr, a_kwarg=10):
+            def impl(arr, a_kwarg=10):
+                return a_kwarg
+            return impl
+
+        @njit
+        def bar(A):
+            return A.foo(), A.foo(20), A.foo(a_kwarg=30)
+
+        Z = np.arange(5)
+
+        self.assertEqual(bar(Z), (10, 20, 30))
+
+    @unittest.skipUnless(_IS_PY3, "unicode not supported in py2")
```

Comment:
Would substituting `StringLiteral` for `IntegerLiteral` use permit this test to also work on python 2.7?

Reply:
True.  I was not thinking.  I'll do that.